--- a/libs/love/CMakeLists.txt
+++ b/libs/love/CMakeLists.txt
@@ -2028,6 +2028,8 @@ else()
                        "-lidbfs.js" # to fix the ReferenceError: IDBFS is not defined
                        "-lwebsocket.js" # websocket API
                        "-lopenal"
+                       "${CMAKE_SOURCE_DIR}/libffi.a"
+                       "${CMAKE_SOURCE_DIR}/libcffi-lua.a"
                )

                add_executable(${LOVE_EXE_NAME} src/love.cpp)
